src/mender-update/daemon/states.hpp

```diff
@@ -58,25 +58,34 @@ class IdleState : virtual public StateType {
 
 class PollForDeploymentState : virtual public StateType {
 public:
-	PollForDeploymentState(events::EventLoop &loop) :
-		poll_timer_(loop) {
-	}
+	PollForDeploymentState(
+		events::EventLoop &event_loop, int retry_interval_seconds, int retry_count);
+
 	void OnEnter(Context &ctx, sm::EventPoster<StateEvent> &poster) override;
 
 private:
-	events::Timer poll_timer_;
+	void HandlePollingError(Context &ctx, sm::EventPoster<StateEvent> &poster);
+	struct Retry {
+		http::ExponentialBackoff backoff;
+		events::Timer wait_timer;
+	};
+	optional<Retry> retry_;
```

Comment:
Is it really optional? I think the state will miss-behave if it is not initialized with the retry

Reply:
Yeah, I don't think it should be optional - left over from when I ported the backoff here